src/main/java/org/intellij/sdk/codesync/HandleBuffer.java

```diff
@@ -0,0 +1,313 @@
+     */
+    public static  DiffFile[] getDiffFiles(String path, String diffFileExtension)  {
+        File diffFilesDirectory = new File(path);
+
+        File[] files = diffFilesDirectory.listFiles(
+            (dir, name) -> name.toLowerCase().endsWith(diffFileExtension.toLowerCase())
+        );
+        if (files != null)  {
+            return Arrays.stream(files).map(DiffFile::new).toArray(DiffFile[]::new);
+        }
+        return new DiffFile[0];
+    }
+
+    public static void handleBuffer() {
+        ConfigFile configFile;
+        HashSet<String> newFiles = new HashSet<>();
+
+        DiffFile[] diffFiles = getDiffFiles(DIFFS_REPO, ".yml");
+        if (diffFiles.length == 0) {
+            return;
+        }
+
+        CodeSyncClient client =new CodeSyncClient();
+        if (!client.isServerUp()) {
+            return;
+        }
+
+        diffFiles = Arrays.copyOfRange(
+            diffFiles, 0, diffFiles.length >= DIFFS_PER_ITERATION ? DIFFS_PER_ITERATION : diffFiles.length
+        );
+
+        try {
+            configFile = new ConfigFile(CONFIG_PATH);
+        } catch (InvalidConfigFile error) {
+            System.out.printf("Config file error, %s%n", error.getMessage());
+            return;
+        }
+
+        for (DiffFile diffFile: diffFiles) {
+            if (!diffFile.isValid()) {
+                String filePath =  String.format("%s/%s", diffFile.repoPath, diffFile.fileRelativePath );
+                System.out.printf("Skipping invalid diff file: %s. data: %s", filePath, diffFile.diff);
+
+                diffFile.delete();
+                continue;
+            }
+
+            if (!configFile.repos.containsKey(diffFile.repoPath)) {
+                System.out.printf("Repo `%s` is in buffer.yml but not in configFile.yml", diffFile.repoPath);
+                continue;
+            }
+
+            ConfigRepo configRepo = configFile.getRepo(diffFile.repoPath);
+            if (!configRepo.branches.containsKey(diffFile.branch))  {
+                System.out.printf("Branch: `%s` is not synced for Repo `%s`", diffFile.branch, diffFile.repoPath);
+                continue;
+            }
+
+            WebSocketClient webSocketClient;
+            try {
+                webSocketClient = client.connectWebSocket();
+            } catch (WebSocketConnectionError error) {
+                System.out.printf("Failed to connect to websocket endpoint: %s", WEBSOCKET_ENDPOINT);
+                continue;
+            }
+
+            Boolean authenticated = webSocketClient.authenticate(configRepo.token);
+            if (!authenticated) {
+                System.out.println("Could noy authenticate user with token");
+                continue;
+            }
+
+            if(diffFile.isDirRename) {
+                handleDirRename(diffFile);
+                diffFile.delete();
+                continue;
+            }
+
+            if (Utils.shouldIgnoreFile(diffFile.fileRelativePath, diffFile.repoPath)) {
```

Comment:
@saleem-latif This is good but I skipped this check in JS side as diffs are only computed/added if file is syncable. This can help if someone manully edits a diff file for some non-syncable file. 

I will add in JS side if issue arises.